### PR TITLE
Add TSVConnFdGet api

### LINKED ISSUE #10323: Create helper TS API to return the file descriptor associated with a SSL connection
I needed this to simplify writing a plugin that needed some access to the SSL object information.  For a variety of reasons, I didn't want to pull in a SSL library to fetch this information directly from the SSL object I could have retrieved from TSVConnSslConnectionGet.

I also had created a wrapper API to return the SNI, but I see that someone has already contributed that via the TSVConnSslSniGet function.

### PR DESCRIPTION
**Pull request by shinrich**
Adding a convenience API for working with SSL connections.

This closes #10323

### CODE REVIEW

File: doc/developer-guide/api/functions/TSVConnSslFdGet.en.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+.. code-block:: cpp
+
+    #include <ts/ts.h>
+
+.. c:function:: int TSVConnSslFdGet(TSVConn sslp)
+
+
+Description
+-----------
+Returns the file descriptor associated with the SSL connection :arg:`sslp`.
```
Comment (maskit) [inline doc/developer-guide/api/functions/TSVConnSslFdGet.en.rst:33]: Why do we need to limit this just for SSL connection? Are we going to add `TSVConnFdGet` and `TSVConnQuicFdGet` if we needed? Plugins would need to call isSSL() and isQUIC() then call the corresponding TSVConnXFdGet()`. It doesn't sounds handy.

Comment (shinrich) [reply doc/developer-guide/api/functions/TSVConnSslFdGet.en.rst:33]: Yeah, that crossed my mind as I pushed the branch up.  Let me take a pass at generalizing.

File: include/ts/ts.h [hunk 2]
```diff
@@ -1315,6 +1315,8 @@ namespace c
   TSReturnCode TSVConnTunnel(TSVConn sslp);
   /*  Return the SSL object associated with the connection */
   TSSslConnection TSVConnSslConnectionGet(TSVConn sslp);
+  /* Return the file descriptoer associated with the ssl connection */
```
Comment (maskit) [inline include/ts/ts.h:1318]: Please remove "ssl".

**Comment by maskit** [comment]
The change looks almost fine, but I don't see API proposal for this on the list.

File: include/ts/ts.h (hunk 2 above)
Comment (zwoop) [reply include/ts/ts.h:1318]: From the comment :)

**Review by zwoop — CHANGES_REQUESTED** [review]

**Comment by shinrich** [comment]
Sent the API email to the list a couple days ago.  Will make a new PR with the remaining changes next week.

**Review by bneradt — APPROVED** [review]

**Review by zwoop — APPROVED** [review]

**Pull request merged**